doc/concepts.md

```diff
@@ -1,189 +1,264 @@
-Relation to SOFT
-================
-SOFT stands for SINTEF Open Framework and Tools and is a set of
-concepts for how to achieve semantic interoperability as well as
-implementations and tooling.  At the core of SOFT are the [SOFT data
-models], which provide a by design simplistic but powerful way to
-represent scientific data.
-
-DLite is one implementation of the [SOFT data models].  Originally it
-was a simplified pure C implementation of SOFT5, but has with time
-developed into a project of its own.  This document will highlight a
-few notable differences with SOFT5.
-
-
-Named data instances
---------------------
-In SOFT all instances are referred to by their UUID.  All metadata are
-uniquely identified by their URI, which must be of the form
-'`namespace`/`version`/`name`', like:
-http://onto-ns.com/meta/0.3/EntitySchema.  This is also valid in
-DLite, but we allow the URI to optionally end with a hash (#) or slash
-(/), which will be ignored.  Since a metadata is an instance of its
-meta-metadata, it also has an UUID, which is calculated from the URI
-(as an UUID version 5 SHA-1 hash of the URI using the DNS namespace).
-
-In some cases when you have unique and immutable data, e.g. default
-input parameters to a given version of a software model, it may be
-more convenient to refer to an unique human understandable name (URI),
-like "`mymodel-1.2.3-default_input`", rather than a UUID on the form
-"`8290318f-258e-54e2-9838-bb187881f996`".  DLite supports this.
-Currently DLite does not enforce that user-defined URIs must follow the
-[RFC 3986] standard for a [valid URI], but it is recommended to
-follow it.
-
+DLite
+=====
+DLite is an implementation of [SOFT], which stands for SINTEF Open
+Framework and Tools and is a set of concepts for how to achieve
+semantic interoperability as well as implementations and corresponding
+tooling.
+
+The development of SOFT was motivated by many years of experience with
+developing scientific software, where it was observed that a lot of
+efforts went into developing parts that had little to do with the
+domain.
+A significant part of the development process was spent on different
+software engineering tasks, such as code design, the handling of I/O,
+correct memory handling of the program state and writing import and
+export filters in order to use data from different
+sources.
+In addition comes the code maintenance with support of legacy formats
+and the introduction of new features and changes to internal data
+state in the scientific software.
+With SOFT it is possible to utilize reusable software components that
+handle all this, or develop new reusable software components that can
+be used by others in the same framework.
+At the core of SOFT are the [SOFT data models], which provide a by
+design simplistic but powerful way to represent scientific data.
```

Comment:
```suggestion
At the core of SOFT are the [SOFT data models], which by design provide a simple but powerful way to represent scientific data.
```

Reply:
Dont use the word 'simplistic'. It means too simple and not including all that is necessary: https://www.britannica.com/dictionary/simplistic